cmake/toolchain/Kconfig

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config TOOLCHAIN_HAS_PICOLIBC
+	bool
+	help
+	  Toolchain has picolibc library.
+
+config TOOLCHAIN_CAN_BUILD_PICOLIBC_MODULE
```

Comment:
I think the existing Kconfig bits end up assuming that all toolchains can build the picolibc module by default? Maybe this should default to 'y'?

Reply:
`CONFIG_PICOLIBC_SUPPORTED` would only allow Zephyr SDK to build the module. I am simply trying to preserve the current behavior. Since I don't have access to all toolchain, I am not comfortable in flipping the behavior mode.

Reply:
yup, I was just mistaken about what the default behavior was.

Reply:
The definition of `TOOLCHAIN_CAN_BUILD_PICOLIBC_MODULE` should be moved elsewhere.
As the setting is closely coupled to the picolibc module itself, then this could be a good location:
https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/Kconfig.picolibc

A toolchain which supports building picolibc can then do a `select TOOLCHAIN_CAN_BUILD_PICOLIBC_MODULE`, as it is done for `THREAD_LOCAL_STORAGE`.
For example:
https://github.com/zephyrproject-rtos/zephyr/blob/eee70b9d096aebc6ba8dda02efe19526ace6c369/cmake/toolchain/xtools/Kconfig#L4-L6

extra comment: would `TOOLCHAIN_SUPPORTS_PICOLIBC_MODULE` be a better name, as it aligns to other `<something>_SUPPORTS_<name>` var names used in Kconfig ?